src/protocol/resp/src/message/integer.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use super::*;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Integer {
+    pub(crate) inner: u64,
```

Comment:
i think RESP integers are signed 64 bit, not unsigned, according to the protcol spec https://redis.io/docs/reference/protocol-spec/

Reply:
You're right. Fixed